src/test/crimson/seastore/test_extmap_manager.cc

```diff
@@ -0,0 +1,312 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "test/crimson/gtest_seastar.h"
+
+#include "crimson/os/seastore/cache.h"
+#include "crimson/os/seastore/transaction_manager.h"
+#include "crimson/os/seastore/segment_manager.h"
+#include "crimson/os/seastore/extentmap_manager.h"
+
+#include "test/crimson/seastore/test_block.h"
+
+using namespace crimson;
+using namespace crimson::os;
+using namespace crimson::os::seastore;
+
+namespace {
+  [[maybe_unused]] seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_test);
+  }
+}
+
+
+struct extentmap_manager_test_t : public seastar_test_suite_t {
+  std::unique_ptr<SegmentManager> segment_manager;
+  Journal journal;
+  Cache cache;
+  LBAManagerRef lba_manager;
+  TransactionManager tm;
+  ExtentMapManagerRef extmap_manager;
+
+  extentmap_manager_test_t()
+    : segment_manager(create_ephemeral(segment_manager::DEFAULT_TEST_EPHEMERAL)),
+      journal(*segment_manager),
+      cache(*segment_manager),
+      lba_manager(
+	lba_manager::create_lba_manager(*segment_manager, cache)),
+      tm(*segment_manager, journal, cache, *lba_manager),
+      extmap_manager(
+        extentmap_manager::create_extentmap_manager(&tm)) {}
+
+  seastar::future<> set_up_fut() final {
+    return segment_manager->init().safe_then([this] {
+      return tm.mkfs();
+    }).safe_then([this] {
+      return tm.mount();
+    }).handle_error(
+      crimson::ct_error::all_same_way([] {
+	ASSERT_FALSE("Unable to mount");
+      })
+    );
+  }
+
+  seastar::future<> tear_down_fut() final {
+    return tm.close(
+    ).handle_error(
+      crimson::ct_error::all_same_way([] {
+	ASSERT_FALSE("Unable to close");
+      })
+    );
+  }
+
+  using test_extmap_t = std::map<uint32_t, lext_map_val_t>;
+  test_extmap_t test_ext_mappings;
+
+  TestBlockRef alloc_extent(
```

Comment:
Do you actually need to alloc extents here?  I don't think there's any need for the laddrs in the map to be real for the purposes of this test.  If you use fake values, you can create far more entries in the same space and time resulting in a better test.  Tests later on will use real extents, but this one should focus on exercising the tree structure itself.